### DISCUSSION TITLE
DriveBase track target

### DISCUSSION THREAD

**Discussion by pirsquared** [pybricks/support#1853] [Q&A]
I need an equivalent of the Motor class’ track_target but for DriveBase’s straight method. Before I start writing it, is there functionality I may have missed that makes this easier?

**Comment by laurensvalk** [comment]
If you want to drive straight based on the gyro, you could use `use_gyro(True)` and then use `straight` without further changes.

If you want to continuously change the target like when following a colored ball or line, you could do something like the following:

```python
correction = target - drivebase.angle()
gain = 1 # Choose a suitable feedback
drivebase.drive(speed, correction * gain)
```

**Reply by pirsquared** [reply]
This isn't what I was looking for ... BUT this is an answer to another question I had and so thank you.

What I was looking for, I didn't articulate well and I'll make another attempt at a later date when I share what we've done.  For now I'll leave this still vague but maybe better explanation:

We have a robot with a forklift.  We need the tip of the forklift to travel in a circular path in the x-z plane.  Using track_target on the motor controlling the lift, I can properly time the z-coordinate using sine.  I was hoping for a DriveBase method that would allow me to track_distance.  Instead what I did is use track_target on the left and right motors of the DriveBase.  So the left and right motors and moving forward controlled by the cosine function and the lift motor controlled by the sine function and voila, I have a circular arc.

In fact we have 4 motors and we wrote functionality to accept callables that act as parametric equations and are coordinated with time and track_target for each.

As I said, I'll share our results when I code is better formed.
